frontend/src/pages/_app.tsx

```diff
@@ -78,8 +79,28 @@ declare module "next/app" {
   }
 }
 
+const wrapHandler = <T extends NextRouter["replace"]>(
+  router: NextRouter,
+  originalHandler: T
+): T => {
+  return ((href, as, ...args) => {
+    const network = router.query.network;
+    if (network) {
+      href += `?network=${network}`;
```

Comment:
What do you think of this hacky solution guys?
Should I parse `href` and `as` and append network properly? Or it is good enough until we have more query params?